management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -66,6 +66,34 @@ public synchronized void unregister()
         RpcRegistry.unregister(RPC_CLASS_NAME);
     }
 
+    @Rpc(name = "setFullQuerylog")
+    public void setFullQuerylog(@RpcParam(name="enabled") boolean fullQueryLoggingEnabled) throws UnsupportedOperationException
+    {
+        logger.debug("Attempting to set full query logging to " + fullQueryLoggingEnabled);
+        String cassVersion = ShimLoader.instance.get().getStorageService().getReleaseVersion();
+        if (Integer.parseInt(cassVersion.split("\\.")[0]) < 4){
+            logger.error("Full query logging is not available in Cassandra < 4x. This call is going to fail.");
+        }
+        // Warning: bad design here. Implementations for 3.x and DSE will throw UnsupportedOperationException at runtime.
```

Comment:
nit: You can remove this comment now. Or maybe move it p above the Cassandra 4.0+ check above?

Reply:
Done.